### ISSUE TITLE
Creating a new image with default data 

### ISSUE THREAD

**Issue by mlaro** [sclorg/postgresql-container#285]
HI, 
We have some bootstrap data  that we would like to have the container always have at a minimum.  

Is there a way to modify the image so that it can retain this initial datafill?  

Commit does not seem to work .

thanks Mike

**Comment by mlaro** [comment]
help .... the container seems to always re-initialize the DB

**Comment by pkubatrh** [comment]
Hi Mike,

I do not think this specific use case is properly documented but you can certainly do that with s2i. You just need to provide the initial data and a custom assemble script that loads in the data when creating the image with s2i.

For an actual example on how to do this you can check out this test case:
https://github.com/sclorg/postgresql-container/blob/master/test/run_test#L711

**Comment by mlaro** [comment]
I am not really familiar with s2i. I will take a look.

I really was hoping for an easy way to just add my PostgreSQL data
directory and stop the container startup from wiping it out.

Thanks so much for the reply

**Comment by pkubatrh** [comment]
As for why `docker commit` did not work for you I think this is caused by the use of the `VOLUME` command in our dockerfiles. Docker likely does not include any volumes it creates into the image when the container gets committed.

**Comment by mlaro** [comment]
Yes ... you are correct.  Is there an easy way to mv (ADD) the PostgreSQL
PGDATA directory and contents into the container through a Dockerfile and
stop the default startup of this from replacing the contents ?

Thanks so much for your help

**Comment by praiskup** [comment]
If you are running on OpenShift, the easiest thing is to use the s2i as @pkubatrh mentioned above (maybe look at slightly clearer [link to example s2i app](https://github.com/sclorg/postgresql-container/tree/master/examples/s2i-dump-data)).

If s2i is not what you want, create the Dockerfile manually... Then in Dockerfile `COPY`  the data directory somewhere else then `/var/lib/pgsql/data`, and move it on place by `CMD` at container startup.

**Comment by mlaro** [comment]
Thanks Pavel,

I guess it's not clear, or I did not find what I need to do to
the centos/postgresql-96-centos7 image to execute the s2i hook.

**Comment by mlaro** [comment]
Hmmm ... I tried using the Dockerfile and CMD to move the data dir ... it
works I think .... but the container exits after the CMD to move

**Comment by praiskup** [comment]
On Thursday, August 2, 2018 3:43:45 PM CEST Mike wrote:
> I guess it's not clear, or I did not find what I need to do to
> the centos/postgresql-96-centos7 image to execute the s2i hook.

Take a look at s2i page homepage (plain docker):
https://github.com/openshift/source-to-image

Or at OpenShift documentation (OpenShift use-case):
https://github.com/openshift/source-to-image

**Comment by praiskup** [comment]
Note to myself, we should add `README` into https://github.com/sclorg/postgresql-container/tree/master/examples/s2i-dump-data (at least) explaining the basic "howto".
